src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/jobsboard/api/ExpressionOfInterestRepositoryShould.kt

```diff
@@ -0,0 +1,306 @@
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.whenever
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.context.annotation.Import
+import org.springframework.dao.DataAccessException
+import org.springframework.data.auditing.DateTimeProvider
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.jobsboard.api.commons.domain.EntityId
+import uk.gov.justice.digital.hmpps.jobsboard.api.config.TestJpaConfig
+import uk.gov.justice.digital.hmpps.jobsboard.api.employers.domain.Employer
+import uk.gov.justice.digital.hmpps.jobsboard.api.employers.domain.EmployerRepository
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterest
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterestId
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterestRepository
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.Job
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.JobRepository
+import uk.gov.justice.digital.hmpps.jobsboard.api.testcontainers.PostgresContainer
+import java.time.Instant
+import java.time.LocalDate
+import java.time.Month.JULY
+import java.util.*
+
+@DataJpaTest
+@Import(TestJpaConfig::class)
+@AutoConfigureTestDatabase(replace = NONE)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ActiveProfiles("test-containers")
+@Transactional
+class ExpressionOfInterestRepositoryShould {
+
+  @Autowired
+  private lateinit var dateTimeProvider: DateTimeProvider
+
+  @Autowired
+  private lateinit var entityManager: EntityManager
+
+  @Autowired
+  private lateinit var employerRepository: EmployerRepository
+
+  @Autowired
+  private lateinit var jobRepository: JobRepository
+
+  @Autowired
+  private lateinit var expressionOfInterestRepository: ExpressionOfInterestRepository
+
+  private val jobCreationTime = Instant.parse("2024-01-01T00:00:00Z")
+  private val jobRegisterExpressionOfInterestTime = Instant.parse("2025-03-01T01:00:00Z")
+  private val jobReRegisterExpressionOfInterestTime = Instant.parse("2025-03-02T01:00:00Z")
+
+  private val expectedPrisonNumber = Holder.expectedPrisonNumber
+  private val amazonForkliftOperatorJob = Holder.amazonForkliftOperatorJob
+  private val amazonEmployer = Holder.amazonEmployer
+  private val nonExistentJob = Holder.nonExistentJob
+
+  private object Holder {
```

Comment:
Is this an attempt to encapsulate all these functions and values to later move them into a mother object?

Reply:
holder of values that we can pick to expose (e.g. to child class when we pull stubs up to parent class) or hide